chain/syncer_integration_test.go

```diff
@@ -27,400 +27,78 @@ import (
-	forklink2 := th.RequireNewTipSet(t, forklink2blk1, forklink2blk2, forklink2blk3)
+	assert.True(t, newStore.HasTipSetAndState(ctx, left.Key().String()))
+	assert.False(t, newStore.HasTipSetAndState(ctx, right.Key().String()))
 
-	fakeChildParams.Nonce = uint64(0)
-	fakeChildParams.Parent = forklink2
-	forklink3blk1 := th.RequireMkFakeChild(t, fakeChildParams)
+	newRight := builder.AppendManyOn(2, right)
 
-	fakeChildParams.Nonce = uint64(1)
-	forklink3blk2 := th.RequireMkFakeChild(t, fakeChildParams)
-	forklink3 := th.RequireNewTipSet(t, forklink3blk1, forklink3blk2)
-
-	_ = requirePutBlocks(t, blockSource, forklink1.ToSlice()...)
-	_ = requirePutBlocks(t, blockSource, forklink2.ToSlice()...)
-	forkHead := requirePutBlocks(t, blockSource, forklink3.ToSlice()...)
-	err = syncer.HandleNewTipset(ctx, forkHead, peer.ID(""))
-	require.NoError(t, err)
-	requireHead(t, chainStore, forklink3)
-
-	// Put blocks in global IPLD blockstore
-	// TODO #2128 make this cleaner along with broad test cleanup.
-	bs := bstore.NewBlockstore(r.Datastore())
-	cst := &hamt.CborIpldStore{Blocks: bserv.New(bs, offline.Exchange(bs))}
-	requirePutBlocksToCborStore(t, cst, dstP.genTS.ToSlice()...)
-	requirePutBlocksToCborStore(t, cst, dstP.link1.ToSlice()...)
-	requirePutBlocksToCborStore(t, cst, dstP.link2.ToSlice()...)
-	requirePutBlocksToCborStore(t, cst, forklink1.ToSlice()...)
-	requirePutBlocksToCborStore(t, cst, forklink2.ToSlice()...)
-	requirePutBlocksToCborStore(t, cst, forklink3.ToSlice()...)
-
-	// Shut down store, reload and wire to syncer.
-	loadSyncer, blockSource := loadSyncerFromRepo(t, r, dstP)
-
-	// Test that the syncer can't sync a block on the old chain
-	// without getting old blocks from network. i.e. the repo is trimmed
+	// Test that the syncer can't sync a block on the right (shorter) chain
+	// without getting old blocks from network. i.e. the store index has been trimmed
 	// of non-heaviest chain blocks
-	cids3 := requirePutBlocks(t, blockSource, dstP.link3.ToSlice()...)
-	err = loadSyncer.HandleNewTipset(ctx, cids3, peer.ID(""))
+	err = offlineSyncer.HandleNewTipset(ctx, newRight.Key(), "")
 	assert.Error(t, err)
-
-	// Test that the syncer can sync a block on the heaviest chain
-	// without getting old blocks from the network.
-	fakeChildParams.Parent = forklink3
-	forklink4blk1 := th.RequireMkFakeChild(t, fakeChildParams)
-	forklink4 := th.RequireNewTipSet(t, forklink4blk1)
-	cidsFork4 := requirePutBlocks(t, blockSource, forklink4.ToSlice()...)
-	err = loadSyncer.HandleNewTipset(ctx, cidsFork4, peer.ID(""))
-	assert.NoError(t, err)
 }
 
 // Syncer handles MarketView weight comparisons.
 // Current issue: when creating miner mining with addr0, addr0's storage head isn't found in the blockstore
 // and I can't figure out why because we pass in the correct blockstore to createStorageMinerWithpower.
 func TestTipSetWeightDeep(t *testing.T) {
-	tf.UnitTest(t)
+	// This test takes many seconds, not sure exactly why.
```

Comment:
Doing a little bit of Printf investigation it looks like the call to gengen takes the majority of time.  Wrapping the gengen call with "start" and "done" print messages the "start" message shows up as soon as the go test output logs `CONT  TestTipSetWeightDeep`.  The output then hangs for 5 seconds and as soon as "done" gets printed `--- PASS: TestTipSetWeightDeep ([5-6]s)` gets printed.

I'm confident enough in this little inspection to recommend replacing "not sure exactly why" with "gengen call dominates/is bottleneck"

Reply:
iirc this is related to gengen allocating power to the miners it defines in the genesis block.